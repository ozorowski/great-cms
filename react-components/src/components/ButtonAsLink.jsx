import React from 'react'
import { node, string } from 'prop-types'
import'./stylesheets/ButtonAsLink.scss'

const ButtonAsLink = ({ children, location }) => (
    <a className="buttonAsLink" href={location}>{children}</a>
)

ButtonAsLink.propTypes = {
    children: node.isRequired,
    location: string.isRequired,
}

export default ButtonAsLink
