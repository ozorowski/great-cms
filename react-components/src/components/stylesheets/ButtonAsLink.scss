@import 'helpers/colours';
// TODO: remove once elements/buttons doesn't error
// @import 'elements/buttons';

.buttonAsLink {
    // @extend .g-button-ghost-white;
    background-color: $grey;
    color: $white;
    display: inline-block;
    font-size: 16px;
    font-weight: bold;
    height: 40px;
    padding: 8px 24px;
    text-decoration: none;
    white-space: nowrap;

    &:hover {
        background-color: darken($grey, 5);
        color: $white;
    }
}
