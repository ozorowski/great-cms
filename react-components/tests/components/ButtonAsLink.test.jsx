import React from 'react'
import { shallow } from 'enzyme'
import ButtonAsLink from '@src/components/ButtonAsLink'

describe('ButtonAsLink component', () => {
    test('should render proper markup', () => {
        const wrapper = shallow(<ButtonAsLink location="moon">Click here</ButtonAsLink>);
        expect(wrapper).toMatchSnapshot();
    })
})
