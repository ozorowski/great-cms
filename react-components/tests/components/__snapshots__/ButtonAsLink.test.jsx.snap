// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAsLink component should render proper markup 1`] = `ShallowWrapper {}`;
